tools/tertiary-analysis/garnett/garnett_train_classifier.xml

```diff
@@ -5,12 +5,40 @@
     </macros>
     <expand macro="requirements" />
     <command detect_errors="exit_code"><![CDATA[
-        garnett_train_classifier.R --cds-object '${cds_object}' --marker-file-path '${marker_file_path}' --database '${database}' --cds-gene-id-type '${cds_gene_id_type}' --marker-file-gene-id-type '${marker_file_gene_id_type}' --num-unknown '${num_unknown}' --min-observations '${min_observations}' --max-training-samples '${max_training_samples}' --propogate-markers '${propogate_markers}' --cores '${cores}' --classifier-gene-id-type '${classifier_gene_id_type}' --output-path '${output_classifier_path}'
+        garnett_train_classifier.R --cds-object '${cds_object}' --marker-file-path '${marker_file_path}' --database '${database}' --output-path '${output_classifier_path}'
+        #if $train_id
+        --train-id '${train_id}' 
+        #end if
+        #if $cds_gene_id_type
+        --cds-gene-id-type '${cds_gene_id_type}' 
+        #end if
+        #if $marker_file_gene_id_type
+        --marker-file-gene-id-type '${marker_file_gene_id_type}' 
+        #end if
+        #if $num_unknown
+        --num-unknown '${num_unknown}' 
+        #end if
+        #if $min_observations
+        --min-observations '${min_observations}' 
+        #end if
+        #if $max_training_samples
+        --max-training-samples '${max_training_samples}' 
+        #end if
+        #if $propogate_markers
+        --propogate-markers '${propogate_markers}'
+        #end if
+        #if $cores 
+        --cores '${cores}' 
```

Comment:
Cores actually shouldn't be set by the user, but the admin through setup decides how many cores are given to a tool execution. To pass what the admin has set for this tool to the tool you should use `$GALAXY_SLOTS` instead like this:

https://github.com/galaxyproject/tools-iuc/blob/20cfde679c4064cea8a0b64763ddd5e092af7eab/data_managers/data_manager_build_kraken_database/data_manager/kraken_database_builder.xml#L15

Reply:
Please do so on every use of cores in the other wrappers.

Reply:
and hence you need to remove $cores from the <inputs> section.

Reply:
Thanks for fixing this!